operator/api/redpanda/v1alpha2/user_types.go

```diff
@@ -129,6 +129,8 @@ type Password struct {
 	Value string `json:"value,omitempty"`
 	// ValueFrom specifies a source for a password to be fetched from when specifying or generating user credentials.
 	ValueFrom *PasswordSource `json:"valueFrom"`
+	// NoGenerate when set to true does not create kubernetes secret when ValueFrom points to none-existent secret.
+	NoGenerate *bool `json:"noGenerate,omitempty"`
```

Comment:
A `*bool` has 3 states (true, false, nil) but we're only using 2 as `nil` is always defaulted to `false`. As `false` is the default value for `bool`, let's just use a `bool`.

Reply:
The problem is with how CRD are displayed for end user. If I leave bool type not a pointer it will always be returned in e.g. `kubectl get user -o yaml` even if user does not set it. That's why I'm opting to hide this. I'm not opposed to your suggestion, but that was my intention.

Reply:
I changed as suggested.